.github/workflows/static.yml

```diff
@@ -0,0 +1,73 @@
+# Workflow to deploy main and staging content to GitHub Pages
+name: Deploy main and staging content to GitHub Pages
+
+on:
+  push:
+    branches:
+      - "main"
+      - "staging"
+
+  workflow_dispatch:
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+
+    steps:
+      # Fetch and prepare main branch content
+      - name: Fetch main branch content into temporary directory
+        run: |
+          echo "Fetching main branch content..."
+          mkdir -p temp_main
+          git clone --branch main --depth 1 https://github.com/${{ github.repository }} temp_main
+          echo "Copying main branch content to deploy directory..."
+          mkdir -p deploy
+          rsync -av --exclude='.git' temp_main/ deploy/
+
+      # Check if staging branch exists before attempting to fetch
+      - name: Check if staging branch exists
```

Comment:
It seems to me like the staging branch should always exist, and we should safely assume that it does? Not sure if that helps reducing some of this code - but also fine to do the check and leave this here.

Reply:
I was considering a scenario (too rare) where the staging branch diverges significantly from the main branch, and a non-dev might prefer to make simpler edits directly through the GitHub UI. In such cases, they could delete the staging branch and recreate it from the main branch. This check could help preempt job failures. Can remove it if we are over-engineering at this point.

Reply:
Haa :) No I mean if it all seems to work, this all makes sense - some comments in the code might help, but honestly this is great - it does make sense to check this edge case I think.